### DISCUSSION TITLE
SD card init failed

### DISCUSSION THREAD

**Discussion by gwgittrash** [jomjol/AI-on-the-edge-device#3461] [General]
### The Problem

I try to install the AI on the Edge Software on my ESP 32. Sadly I always have a problem with die ini-process of the SD Card, ends up in "SD card init failed. Check if SD card is properly inserted into SD card slot or try another card"
I tried 3 different SD Cards, all with 8GB. All formated as FAT32, one partition. I also checked to Note here:
https://jomjol.github.io/AI-on-the-edge-device-docs/Installation/
And followed the discussion here: https://github.com/jomjol/AI-on-the-edge-device/issues/2858
I tried manual installation of [16.0.0-RC5], [15.7.0] and [15.4.0] as also the automated installation of 15.7.0....
What can I do? 

### Version

15.7.0

### Logfile

```shell
rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:3540
load:0x40078000,len:14712
load:0x40080400,len:4
load:0x40080404,len:3096
entry 0x40080584
I (569) quad_psram: This chip is ESP32-D0WD
I (569) esp_psram: Found 8MB PSRAM device
I (569) esp_psram: Speed: 40MHz
I (569) esp_psram: PSRAM initialized, cache is in low/high (2-core) mode.
W (575) esp_psram: Virtual address not enough for PSRAM, map as much as we can. 4MB is mapped
I (583) cpu_start: Multicore app
I (1467) esp_psram: SPI SRAM memory test OK
I (1476) cpu_start: Pro cpu start user code
I (1476) cpu_start: cpu freq: 160000000 Hz
I (1476) app_init: Application information:
I (1476) app_init: Project name:     AI-on-the-edge
I (1481) app_init: App version:      v16.0.0-RC5
I (1485) app_init: Compile time:     Dec  5 2024 00:08:47
I (1490) app_init: ELF file SHA256:  a94a67001...
I (1495) app_init: ESP-IDF:          5.3.1
I (1498) efuse_init: Min chip rev:     v0.0
I (1502) efuse_init: Max chip rev:     v3.99 
I (1506) efuse_init: Chip rev:         v3.0
I (1511) heap_init: Initializing. RAM available for dynamic allocation:
I (1517) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (1522) heap_init: At 3FFBB4F8 len 00024B08 (146 KiB): DRAM
I (1527) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (1533) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (1538) heap_init: At 4009AC60 len 000053A0 (20 KiB): IRAM
I (1544) esp_psram: Adding pool of 4081K of PSRAM memory to heap allocator
I (1552) spi_flash: detected chip: gd
I (1553) spi_flash: flash io: dio
W (1558) i2c: This driver is an old driver, please migrate your application code to adapt `driver/i2c_master.h`
I (1578) main_task: Started on CPU0
I (1588) esp_psram: Reserving pool of 40K of internal memory for DMA/internal allocations
I (1588) main_task: Calling app_main()
I (1588) MAIN: 



================ Start app_main =================
E (1628) sdmmc_common: sdmmc_init_ocr: send_op_cond (1) returned 0x107
E (1628) vfs_fat_sdmmc: sdmmc_card_init failed (0x107).
E (1628) MAIN: SD card init failed. Check if SD card is properly inserted into SD card slot or try another card
E (1638) MAIN: Device init aborted!
I (1648) main_task: Returned from app_main()
```

**Comment by gwgittrash** [comment]
Additional Info... 
It is an ESP32-CAM developer board ([Link](https://de.aliexpress.com/item/1005007894548215.html?spm=a2g0o.order_list.order_list_main.23.23345c5fYZ2QMq&gatewayAdapt=glo2deu))
I also tried it with and without connected camera...

**Comment by SybexX** [comment]
It looks like an external 10k pullup is missing. There should be five pullups, but only four can be seen in the pictures.

**Comment by gwgittrash** [comment]
Hi and thank you for your support. 
Flashing was successful but the error is still the same
![image](https://github.com/user-attachments/assets/b4f8a62e-619c-4061-82f1-203b580de920)

`ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:3540
load:0x40078000,len:14712
load:0x40080400,len:4
load:0x40080404,len:3096
entry 0x40080584
I (570) quad_psram: This chip is ESP32-D0WD
I (570) esp_psram: Found 8MB PSRAM device
I (571) esp_psram: Speed: 40MHz
I (571) esp_psram: PSRAM initialized, cache is in low/high (2-core) mode.
W (576) esp_psram: Virtual address not enough for PSRAM, map as much as we can. 4MB is mapped
I (584) cpu_start: Multicore app
I (1467) esp_psram: SPI SRAM memory test OK
I (1476) cpu_start: Pro cpu start user code
I (1477) cpu_start: cpu freq: 160000000 Hz
I (1477) app_init: Application information:
I (1477) app_init: Project name:     AI-on-the-edge
I (1481) app_init: App version:      d1e1c67
I (1485) app_init: Compile time:     Dec 29 2024 10:49:50
I (1491) app_init: ELF file SHA256:  bb2fd2bdc...
I (1495) app_init: ESP-IDF:          5.3.1
I (1499) efuse_init: Min chip rev:     v0.0
I (1503) efuse_init: Max chip rev:     v3.99 
I (1507) efuse_init: Chip rev:         v3.0
I (1511) heap_init: Initializing. RAM available for dynamic allocation:
I (1517) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (1522) heap_init: At 3FFBB530 len 00024AD0 (146 KiB): DRAM
I (1528) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (1533) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (1539) heap_init: At 4009AC60 len 000053A0 (20 KiB): IRAM
I (1545) esp_psram: Adding pool of 4081K of PSRAM memory to heap allocator
I (1552) spi_flash: detected chip: gd
I (1554) spi_flash: flash io: dio
W (1558) i2c: This driver is an old driver, please migrate your application code to adapt `driver/i2c_master.h`
I (1579) main_task: Started on CPU0
I (1589) esp_psram: Reserving pool of 40K of internal memory for DMA/internal allocations
I (1589) main_task: Calling app_main()
I (1589) MAIN: 



================ Start app_main =================
E (1729) sdmmc_common: sdmmc_init_ocr: send_op_cond (1) returned 0x107
E (1729) vfs_fat_sdmmc: sdmmc_card_init failed (0x107).
E (1729) MAIN: SD card init failed. Check if SD card is properly inserted into SD card slot or try another card
E (1739) MAIN: Device init aborted!
I (1749) main_task: Returned from app_main()`

**Comment by SybexX** [comment]
Do you have a wiring diagram for the board to see if the pinouts even match, I couldn't quickly find one online.

**Comment by gwgittrash** [comment]
No, I don't have a wiring diagram. But I made some better pictures for you. maybe that helps?
![20241229_124553](https://github.com/user-attachments/assets/6e991cfd-1644-485f-98b8-13936dad63f6)
![20241229_124537](https://github.com/user-attachments/assets/add726da-8b71-4854-b056-aa32a697e5d1)

**Comment by SybexX** [comment]
It looks like the gpios are assigned differently, since I found several for the same board and I don't know which one is right, you have to contact the seller or find out by trying around^^

**Comment by gwgittrash** [comment]
I tried the version fron above, but still the same error. In the meandtime I asked the seller to provide a wiring diagram but I also found this thread: https://community.home-assistant.io/t/esp32-camera-ov5640/507409
Could that be helpful?

**Comment by gwgittrash** [comment]
Additional Information: 
The chip is an ESP32-DOWDQ6-V3, here you will finde the datasheet:
https://www.alldatasheet.com/datasheet-pdf/view/1243004/ESPRESSIF/ESP32-DOWDQ6-V3.html 
Which pins are relevant? maybe I could find out more...

**Comment by gwgittrash** [comment]
Still no success.. Tried with three 8GB cards.. :( 

`ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:3540
load:0x40078000,len:14712
load:0x40080400,len:4
load:0x40080404,len:3096
entry 0x40080584
I (570) quad_psram: This chip is ESP32-D0WD
I (570) esp_psram: Found 8MB PSRAM device
I (571) esp_psram: Speed: 40MHz
I (571) esp_psram: PSRAM initialized, cache is in low/high (2-core) mode.
W (576) esp_psram: Virtual address not enough for PSRAM, map as much as we can. 4MB is mapped
I (584) cpu_start: Multicore app
I (1468) esp_psram: SPI SRAM memory test OK
I (1477) cpu_start: Pro cpu start user code
I (1477) cpu_start: cpu freq: 160000000 Hz
I (1477) app_init: Application information:
I (1477) app_init: Project name:     AI-on-the-edge
I (1482) app_init: App version:      1482a55
I (1486) app_init: Compile time:     Dec 29 2024 15:17:04
I (1491) app_init: ELF file SHA256:  9253d8ded...
I (1495) app_init: ESP-IDF:          5.3.1
I (1499) efuse_init: Min chip rev:     v0.0
I (1503) efuse_init: Max chip rev:     v3.99 
I (1507) efuse_init: Chip rev:         v3.0
I (1511) heap_init: Initializing. RAM available for dynamic allocation:
I (1517) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (1523) heap_init: At 3FFBB530 len 00024AD0 (146 KiB): DRAM
I (1528) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (1533) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (1539) heap_init: At 4009AC60 len 000053A0 (20 KiB): IRAM
I (1545) esp_psram: Adding pool of 4081K of PSRAM memory to heap allocator
I (1552) spi_flash: detected chip: gd
I (1554) spi_flash: flash io: dio
W (1559) i2c: This driver is an old driver, please migrate your application code to adapt `driver/i2c_master.h`
I (1579) main_task: Started on CPU0
I (1589) esp_psram: Reserving pool of 40K of internal memory for DMA/internal allocations
I (1589) main_task: Calling app_main()
I (1589) MAIN: 



================ Start app_main =================
E (1709) vfs_fat_sdmmc: slot init failed (0x102).
E (1709) MAIN: SD card init failed. Check error code or try another card
E (1709) MAIN: Device init aborted!
I (1709) main_task: Returned from app_main()`

**Comment by SybexX** [comment]
It's about these GPIOs, either something doesn't fit or because the IP5306 is connected to the two GPIOs.
![IO-Configuration2](https://github.com/user-attachments/assets/4cdd8fee-396a-4efe-b646-b8529069a812)

https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/error-codes.html

**Reply by gwgittrash** [reply]
Sadly still the same...
I hope I get an answer soon fron the seller...


`ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:3540
load:0x40078000,len:14712
load:0x40080400,len:4
load:0x40080404,len:3096
entry 0x40080584
I (570) quad_psram: This chip is ESP32-D0WD
I (571) esp_psram: Found 8MB PSRAM device
I (571) esp_psram: Speed: 40MHz
I (571) esp_psram: PSRAM initialized, cache is in low/high (2-core) mode.
W (576) esp_psram: Virtual address not enough for PSRAM, map as much as we can. 4MB is mapped
I (584) cpu_start: Multicore app
I (1468) esp_psram: SPI SRAM memory test OK
I (1477) cpu_start: Pro cpu start user code
I (1477) cpu_start: cpu freq: 160000000 Hz
I (1477) app_init: Application information:
I (1477) app_init: Project name:     AI-on-the-edge
I (1482) app_init: App version:      56c3312
I (1486) app_init: Compile time:     Dec 29 2024 17:55:31
I (1491) app_init: ELF file SHA256:  047b35897...
I (1495) app_init: ESP-IDF:          5.3.1
I (1499) efuse_init: Min chip rev:     v0.0
I (1503) efuse_init: Max chip rev:     v3.99 
I (1507) efuse_init: Chip rev:         v3.0
I (1512) heap_init: Initializing. RAM available for dynamic allocation:
I (1518) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (1523) heap_init: At 3FFBB530 len 00024AD0 (146 KiB): DRAM
I (1528) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (1533) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (1539) heap_init: At 4009AC60 len 000053A0 (20 KiB): IRAM
I (1545) esp_psram: Adding pool of 4081K of PSRAM memory to heap allocator
I (1552) spi_flash: detected chip: gd
I (1554) spi_flash: flash io: dio
W (1559) i2c: This driver is an old driver, please migrate your application code to adapt `driver/i2c_master.h`
I (1579) main_task: Started on CPU0
I (1589) esp_psram: Reserving pool of 40K of internal memory for DMA/internal allocations
I (1589) main_task: Calling app_main()
I (1589) MAIN: 



================ Start app_main =================
E (1709) vfs_fat_sdmmc: slot init failed (0x102).
E (1709) MAIN: SD card init failed. Check error code or try another card
E (1709) MAIN: Device init aborted!
I (1709) main_task: Returned from app_main()`

**Reply by SybexX** [reply]
I'm assuming that they changed the pin assignment in rev1.2.1 compared to v1.2.
And the picture above is from v1.2

This is another variant of v1.2 that I could find on the internet^^
![IO-Configuration2_2](https://github.com/user-attachments/assets/e6de06bf-8d90-4ca3-aa01-45f61ad458f0)

And none have worked for you so far^^

**Reply by gwgittrash** [reply]
Hi
I received some information from the seller, but I'm afraid it's nothing new...

[OV5640 ESP32 DevBoard.zip](https://github.com/user-attachments/files/18273249/OV5640.ESP32.DevBoard.zip)
[UserInfo OV5640 ESP32 DevBoard.zip](https://github.com/user-attachments/files/18273251/UserInfo.OV5640.ESP32.DevBoard.zip)

**Reply by SybexX** [reply]
After searching for a while, I found that the MMC mode on the ESP32 only works with the default GPIOs (GPIO_MATRIX only works on the ESP32-S3).

**Reply by gwgittrash** [reply]
I tried it and it seems like you are right (of course :-) )...  
Both are not working, but with different reasons... 

Version 12545561216:
`ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:3540
load:0x40078000,len:14712
load:0x40080400,len:4
load:0x40080404,len:3096
entry 0x40080584
I (577) quad_psram: This chip is ESP32-D0WD
I (577) esp_psram: Found 8MB PSRAM device
I (577) esp_psram: Speed: 40MHz
I (577) esp_psram: PSRAM initialized, cache is in low/high (2-core) mode.
W (583) esp_psram: Virtual address not enough for PSRAM, map as much as we can. 4MB is mapped
I (591) cpu_start: Multicore app
I (1474) esp_psram: SPI SRAM memory test OK
I (1483) cpu_start: Pro cpu start user code
I (1483) cpu_start: cpu freq: 160000000 Hz
I (1483) app_init: Application information:
I (1484) app_init: Project name:     AI-on-the-edge
I (1488) app_init: App version:      97097b6
I (1492) app_init: Compile time:     Dec 30 2024 11:00:59
I (1497) app_init: ELF file SHA256:  2ceffeac8...
I (1502) app_init: ESP-IDF:          5.3.1
I (1506) efuse_init: Min chip rev:     v0.0
I (1510) efuse_init: Max chip rev:     v3.99 
I (1514) efuse_init: Chip rev:         v3.0
I (1518) heap_init: Initializing. RAM available for dynamic allocation:
I (1524) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (1529) heap_init: At 3FFBB5F8 len 00024A08 (146 KiB): DRAM
I (1534) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (1540) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (1546) heap_init: At 4009C1D4 len 00003E2C (15 KiB): IRAM
I (1551) esp_psram: Adding pool of 4081K of PSRAM memory to heap allocator
I (1559) spi_flash: detected chip: gd
I (1561) spi_flash: flash io: dio
W (1565) i2c: This driver is an old driver, please migrate your application code to adapt `driver/i2c_master.h`
I (1585) main_task: Started on CPU0
I (1595) esp_psram: Reserving pool of 40K of internal memory for DMA/internal allocations
I (1595) main_task: Calling app_main()
I (1595) MAIN: 



================ Start app_main =================
**I (1715) MAIN: Mounting filesystem
I (1715) gpio: GPIO[19]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 
E (1755) sdmmc_sd: sdmmc_init_spi_crc: sdmmc_send_cmd_crc_on_off returned 0x106
E (1755) vfs_fat_sdmmc: sdmmc_card_init failed (0x106).
I (1755) gpio: GPIO[19]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 
E (1765) MAIN: SD card init failed. Check error code or try another card**
E (1765) MAIN: Device init aborted!
I (1775) main_task: Returned from app_main()`

Version 12545561216:
`ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:3540
load:0x40078000,len:14712
load:0x40080400,len:4
load:0x40080404,len:3096
entry 0x40080584
I (570) quad_psram: This chip is ESP32-D0WD
I (570) esp_psram: Found 8MB PSRAM device
I (571) esp_psram: Speed: 40MHz
I (571) esp_psram: PSRAM initialized, cache is in low/high (2-core) mode.
W (576) esp_psram: Virtual address not enough for PSRAM, map as much as we can. 4MB is mapped
I (584) cpu_start: Multicore app
I (1468) esp_psram: SPI SRAM memory test OK
I (1477) cpu_start: Pro cpu start user code
I (1477) cpu_start: cpu freq: 160000000 Hz
I (1477) app_init: Application information:
I (1477) app_init: Project name:     AI-on-the-edge
I (1482) app_init: App version:      bbbaa83
I (1486) app_init: Compile time:     Dec 30 2024 10:19:55
I (1491) app_init: ELF file SHA256:  370ee4c92...
I (1495) app_init: ESP-IDF:          5.3.1
I (1499) efuse_init: Min chip rev:     v0.0
I (1503) efuse_init: Max chip rev:     v3.99 
I (1507) efuse_init: Chip rev:         v3.0
I (1511) heap_init: Initializing. RAM available for dynamic allocation:
I (1517) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (1523) heap_init: At 3FFBB530 len 00024AD0 (146 KiB): DRAM
I (1528) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (1533) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (1539) heap_init: At 4009AC60 len 000053A0 (20 KiB): IRAM
I (1545) esp_psram: Adding pool of 4081K of PSRAM memory to heap allocator
I (1552) spi_flash: detected chip: gd
I (1554) spi_flash: flash io: dio
W (1559) i2c: This driver is an old driver, please migrate your application code to adapt `driver/i2c_master.h`
I (1579) main_task: Started on CPU0
I (1589) esp_psram: Reserving pool of 40K of internal memory for DMA/internal allocations
I (1589) main_task: Calling app_main()
I (1589) MAIN: 



================ Start app_main =================
**E (1809) vfs_fat_sdmmc: slot init failed (0x102).
E (1809) MAIN: SD card init failed. Check error code or try another card**
E (1809) MAIN: Device init aborted!
I (1809) main_task: Returned from app_main()`


What does it mean now? Incompatible hardware?

**Reply by SybexX** [reply]
Exactly, this board is not suitable for this project^^

**Reply by gwgittrash** [reply]
Shit... but thank you very very much for you support.
I was very happy to find a model that consists of only one board (no additional cables). But then I'll probably have to keep looking or switch to an alternativ with a spread board...

**Reply by SybexX** [reply]
If the firmware is on the ESP32CAM_AITHINKER, it only needs 5 volts for operation.
I always take an old USB cable and solder it directly to it and for power I use an old cell phone charger.
